framework/src/preconditioners/FieldSplitPreconditioner.C

```diff
@@ -56,20 +56,23 @@ FieldSplitPreconditioner::FieldSplitPreconditioner(const InputParameters & param
   std::unique_ptr<CouplingMatrix> cm = std::make_unique<CouplingMatrix>(n_vars);
   if (!full)
   {
-    const auto off_diag_rows = getParam<std::vector<NonlinearVariableName>>("off_diag_row");
-    const auto off_diag_columns = getParam<std::vector<NonlinearVariableName>>("off_diag_column");
+    if (isParamValid("off_diag_row") && isParamValid("off_diag_column"))
+    {
+      const auto off_diag_rows = getParam<std::vector<NonlinearVariableName>>("off_diag_row");
+      const auto off_diag_columns = getParam<std::vector<NonlinearVariableName>>("off_diag_column");
```

Comment:
is there a check on the size of both somewhere?

Reply:
I couldn't find it. Any hints?

Reply:
maybe add it then? seems like we would be accessing off-bounds

Reply:
This review has been addressed

Reply:
do you mind pointing me to the check?

Reply:
New line 70

Reply:
ok this is not correct.

you ll silently ignore the user inputs if off_diag_row is size 3 and off_diag_col = 0
and you ll still seg fault if
off_diag_row is size 3 and off_diag_col = 1

Reply:
this goes beyond the PR but it d be good to fix. If you want to push this to another PR I m ok with it

Reply:
check the new condition

Reply:
you're ignoring the inputs in case they are not correct.
I think we should have an error if they are not correct.

this check should be right under if (!full)

Reply:
Make an another PR for this then so that we can have proper error messages and tests for error check. I really need this one to merge quickly before any other apps make more changes and I need to patch them again. That's what happens for the Griffin test failure today.

Reply:
other PR sounds good for this item